massa-models/src/amount.rs

```diff
@@ -106,6 +107,25 @@ impl Amount {
         Amount::from_decimal(res)
     }
 
+    /// Creates an amount from a NativeAmount
+    pub fn from_native_amount(amount: &NativeAmount) -> Result<Self, ModelsError> {
```

Comment:
To avoid the low-level Amount to depend on proto-generated stuff, I would recommend doing the conversion through mantissa/scale where it is necessary

Reply:
done